source/elements/oneMKL/source/domains/spblas/data_types/set_dense_matrix_data.rst

```diff
@@ -0,0 +1,176 @@
+.. SPDX-FileCopyrightText: 2024 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. _onemkl_sparse_set_dense_matrix_data:
+
+set_dense_matrix_data
+=====================
+
+Reset the data of a ``dense_matrix_handle_t`` object.
+
+.. rubric:: Description and Assumptions
+
+The ``oneapi::mkl::sparse::set_dense_matrix_data`` function sets new data to the
+``dense_matrix_handle_t`` object with the provided data.
+
+In the case of buffers, the reference count of the provided buffer is
+incremented which extends the lifetime of the underlying buffer until the
+``dmhandle`` is destroyed with ``destroy_dense_matrix`` or the data is reset
+with ``set_dense_matrix_data``.
+
+In the case of USM, the object does not take ownership of the data.
+
+Also see :ref:`onemkl_sparse_create_dense_matrix`.
+
+.. _onemkl_sparse_set_dense_matrix_data_buffer:
+
+set_dense_matrix_data (Buffer version)
+--------------------------------------
+
+.. rubric:: Syntax
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::sparse {
+
+       template <typename dataType>
+       void set_dense_matrix_data (sycl::queue                                &queue,
```

Comment:
If we enforce that `create_dense_matrix` must have all valid values in the arguments so that the handle is 'complete', then we possibly don't need this API.

Reply:
I think it's still useful to let the user reuse the handle and avoid extra allocations.